src/components/Task/MassActions/ConfirmationModal/MassActionsTasksConfirmationModal.tsx

```diff
@@ -20,23 +27,71 @@ export const MassActionsTasksConfirmationModal: React.FC<
 > = ({ open, idsCount, action, setOpen, onConfirm }) => {
   const { t } = useTranslation();
 
+  // KG: useState for checkbox
+  const [checked, setChecked] = useState(false);
+
+  const shouldConfirmDeletion = action === 'delete' && idsCount > 5;
+
+  const handleChange = (event) => {
+    setChecked(event.target.checked);
```

Comment:
You could also write it as `(!checked)` since the new value will always be the opposite of the current one.

```suggestion
    setChecked(!checked)
```